### DISCUSSION TITLE
How do I create a windowless application?

### DISCUSSION THREAD

**Discussion by Local-Trash** [bevyengine/bevy#9209] [Q&A]
I'm learning OpenGL. I wanna learn more, but I also want the ECS of Bevy. I wrote some systems and stuff. When it compiled and ran, it closes in the first frame, because there is no window. Is this possible?

**Comment by Architector4** [comment]
I'm unsure what you mean, but you can indeed use Bevy without a window. That's how `App` object operates by default, if you don't add the `DefaultPlugins`: it has a default runner that runs the app and everything within it once and then returns.

If you ever want to, you can also do `app.set_runner(|mut a| loop { a.update() });` to make it run repeatedly, and then add your initialization one-time-run stuff to `Startup` schedules.

If you want to have most of default plugins (like time and transforms and logging and other stuff) you can either do `app.add_plugins(DefaultPlugins.build().disable::<PluginName>()...)` (repeating the `disable` function) to turn off all the rendering and windowing related plugins, or manually pick out individual plugins you'd like from the default plugins collection here: https://docs.rs/bevy/latest/bevy/struct.DefaultPlugins.html
